src/main/java/impl/ReportCreatorImpl.java

```diff
@@ -0,0 +1,23 @@
+package impl;
+
+import db.Storage;
+import java.util.Map;
+import service.ReportCreator;
+
+public class ReportCreatorImpl implements ReportCreator {
+    private static final String SEPARATOR = ",";
+    private static final String HEADER = "fruit,quantity";
+
+    @Override
+    public String createReport() {
+        StringBuilder report = new StringBuilder(HEADER);
+        report.append(System.lineSeparator());
+        for (Map.Entry<String, Integer> fruits : Storage.fruits.entrySet()) {
```

Comment:
Direct access to `Storage.fruits` breaks encapsulation. It's better to create a method in the `Storage` class that returns a copy of the data or the necessary information without exposing the internal storage directly. This aligns with the encapsulation principle from the checklist.